File: picking-numbers/picking-numbers.js
"use strict";

const fs = require("fs");

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", function (inputStdin) {
  inputString += inputStdin;
});

process.stdin.on("end", function () {
  inputString = inputString.split("\n");

  main();
});

function readLine() {
  return inputString[currentLine++];
}

/*
 * Complete the 'pickingNumbers' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY a as parameter.
 */

function pickingNumbers(a) {
  // Write your code here
  a.sort(function (a, b) {
    return a - b;
  });

  let max = 0;
  let total = 0;
  let diffOne = false;
  for (let i = 0; i < a.length - 1; i++) {
    if (a[i + 1] - a[i] == 0) {
      total++;
    } else if (a[i + 1] - a[i] == 1 && !diffOne) {
      total++;
      diffOne = true;
    } else {
      total = 0;
      diffOne = false;
    }

    if (total > max) max = total;
  }
  return max + 1;
}

function main() {
  const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

  const n = parseInt(readLine().trim(), 10);

  const a = readLine()
    .replace(/\s+$/g, "")
    .split(" ")
    .map((aTemp) => parseInt(aTemp, 10));

  const result = pickingNumbers(a);

  ws.write(result + "\n");

  ws.end();
}

File: picking-numbers/PickingNumbers.java
import java.io.*;
import java.util.*;
import java.util.stream.*;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'pickingNumbers' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY a as parameter.
     */

    public static int pickingNumbers(List<Integer> a) {
        // Write your code here
        Collections.sort(a);
        int max = 0;
        int total = 0;
        boolean diffOne = false;
        for (int i = 0; i < a.size() - 1; i++) {
            if (a.get(i + 1) - a.get(i) == 0) {
                total++;
            } else if (a.get(i + 1) - a.get(i) == 1 && !diffOne) {
                total++;
                diffOne = true;
            } else {
                total = 0;
                diffOne = false;
            }

            if (total > max)
                max = total;
        }
        return max + 1;
    }

}

public class PickingNumbers {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> a = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                .map(Integer::parseInt)
                .collect(toList());

        int result = Result.pickingNumbers(a);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
